front/components/data_source_view/DataSourceViewsSpaceSelector.tsx

```diff
@@ -0,0 +1,81 @@
+import { Spinner } from "@dust-tt/sparkle";
+import type {
+  ContentNodesViewType,
+  DataSourceViewSelectionConfigurations,
+  DataSourceViewType,
+  LightWorkspaceType,
+  SpaceType,
+} from "@dust-tt/types";
+import type { Dispatch, SetStateAction } from "react";
+import { useMemo } from "react";
+
+import { SpaceSelector } from "@app/components/assistant_builder/spaces/SpaceSelector";
+import type { useCaseDataSourceViewsSelector } from "@app/components/data_source_view/DataSourceViewSelector";
+import { DataSourceViewsSelector } from "@app/components/data_source_view/DataSourceViewSelector";
+import { useSpaces } from "@app/lib/swr/spaces";
+
+interface DataSourceViewsSpaceSelectorProps {
+  allowedSpaces?: SpaceType[];
+  owner: LightWorkspaceType;
+  dataSourceViews: DataSourceViewType[];
+  useCase: useCaseDataSourceViewsSelector;
+  selectionConfigurations: DataSourceViewSelectionConfigurations;
+  setSelectionConfigurations: Dispatch<
+    SetStateAction<DataSourceViewSelectionConfigurations>
+  >;
+  viewType: ContentNodesViewType;
+  isRootSelectable: boolean;
+}
+
+export const DataSourceViewsSpaceSelector = ({
+  allowedSpaces,
+  dataSourceViews,
+  useCase,
+  owner,
+  selectionConfigurations,
+  setSelectionConfigurations,
+  viewType,
+  isRootSelectable,
+}: DataSourceViewsSpaceSelectorProps) => {
+  const { spaces, isSpacesLoading } = useSpaces({ workspaceId: owner.sId });
+
+  const defaultSpace = useMemo(() => {
+    const firstKey = Object.keys(selectionConfigurations)[0] ?? null;
+    return firstKey
+      ? selectionConfigurations[firstKey]?.dataSourceView?.spaceId ?? ""
+      : "";
+  }, [selectionConfigurations]);
+
+  if (isSpacesLoading) {
+    return <Spinner />;
+  }
+
+  return (
```

Comment:
That one  should chec the number of spaces. If one single space, return DataSourceViewsSelector directly